src/Facade/Services/ImportBookReportReportFacadeService.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Linn.Stores.Facade.Services
+{
+    using System;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Stores.Domain.LinnApps.ImportBooks;
+    using Linn.Stores.Domain.LinnApps.Reports;
+    using Linn.Stores.Resources.ImportBooks;
+
+    public class ImportBookReportReportFacadeService : IImportBookReportFacadeService
+    {
+        private readonly IImportBookReportService reportService;
+
+        public ImportBookReportReportFacadeService(
+            IRepository<ImportBook, int> repository,
+            IImportBookReportService reportService)
+        {
+            this.reportService = reportService;
+        }
+
+        public IResult<ResultsModel> GetImpbookIPRReport(IPRSearchResource resource)
+        {
+            DateTime from;
+            DateTime to;
+            try
+            {
+                from = DateTime.Parse(resource.FromDate);
+                to = DateTime.Parse(resource.ToDate);
+            }
+            catch (Exception)
+            {
+                return new BadRequestResult<ResultsModel>("Invalid dates supplied to assembly fails details report");
```

Comment:
think this must be a copy paste from assembly fails as should prob say impbook ipr report

Reply:
Yeah sorry, lots of copy and pasting for speed - should have checked through it a bit better though!